.github/workflows/docker.yml

```diff
@@ -8,27 +8,13 @@ on:
     types: [synchronize, opened, reopened, ready_for_review]
 
 jobs:
-  config:
-    runs-on: "ubuntu-latest"
-    outputs:
-      has-secrets: ${{ steps.check.outputs.has-secrets }}
-    steps:
-      - name: "Check for secrets"
-        id: check
-        env:
-          DOCKERHUB_USER: ${{ secrets.DOCKERHUB_USER }}
-          DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
-        run: |
-          if [[ -n "$DOCKERHUB_USER" && -n "$DOCKERHUB_TOKEN" ]]; then
-            echo "has-secrets=true" >> "$GITHUB_ENV"
-            echo "has secrets!"
-          else
-            echo "has-secrets=false" >> "$GITHUB_ENV"
-            echo "no secrets!"
-          fi
+  check-secrets:
+    uses: ./.github/workflows/check-secrets.yml
+    with:
+      secret_name: 'DOCKERHUB_USER'
   docker-build:
-    needs: config
-    if: needs.config.outputs.has-secrets == 'true'
+    needs: check-secrets
+    if: needs.check-secrets.outputs.has-secrets == 'true'
```

Comment:
should we always build images on PRs? like we were doing before? `docker_build_push.sh` will only push if `DOCKERHUB_USER` is present

Reply:
Yes that's right, more work to do here

Reply:
About both your comments, I think I figured it out in https://github.com/apache/superset/pull/26801, where we always run docker, but in forks it runs without authentication, which means it builds an DOESNT push, just making sure that docker build works.

Now I was also able to improve caching there, where in forks, it's use the registry cache as input, but it cannot push the cache and won't. Unclear what the cache hit rate will be, but should be pretty good unless the PR touches dependencies. Also if you don't touch the frontend there's a lot savings on docker build time. 

As docker cache can be really good if done well, it could be tempting to eventually run tests in-docker, and saving a lot on build times.